server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -853,6 +854,8 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
             // TODO: can we avoid having an exception here?
             if (field.getName().equals(IgnoredFieldMapper.NAME)) {
                 builder.field(field.getName(), field.getValues());
+            } else if (field.getName().equals(IgnoredSourceFieldMapper.NAME)) {
```

Comment:
I think this is not needed here. Yaml tests produce output as bytes but that is likely a yaml test issue. I will remove it.